merino/utils/log_data_creators.py

```diff
@@ -94,7 +95,7 @@ def create_suggest_log_data(
         code=message["status"],
         rid=Headers(scope=message)["X-Request-ID"],
         session_id=request.query_params.get("sid"),
-        sequence_no=int(seq) if (seq := request.query_params.get("seq")) else None,
+        sequence_no=int(seq) if (seq.isnumeric() and float(seq).is_integer()) else None,
```

Comment:
Given that `seq` should be any natural number. We can use `isdecimal()` here within the oneliner.
```suggestion
        sequence_no=int(seq) if (seq := request.query_params.get("seq", "")) and seq.isdecimal() else None,
```

Reply:
oooh clever